src/allocator/allocator.c

```
@@ -248,8 +262,10 @@ int remove_chunk_async(
             LIST_REMOVE(a_list,val);
             a_list->limit-=t_size;
             CUdevice dev;
-            cuCtxGetDevice(&dev);
-            rm_gpu_device_memory_usage(getpid(),dev,t_size,2);
+            if (cuCtxGetDevice(&dev) == CUDA_SUCCESS)
+                rm_gpu_device_memory_usage(getpid(), dev, t_size, 2);
+            else
+                LOG_WARN("remove_chunk_async: no current context, skipping accounting");
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)